docs/examples/snippets.c

```diff
@@ -213,9 +213,11 @@ fmi3Status fmi3GetIntervalDecimal(fmi3Instance instance, const fmi3ValueReferenc
 
 	for (int i = 0; i < nValues; i++) {
 		if (valueReferences[i] == 6) {
+            env->lockPreemption(); // Optional: Preventing preemption is actually not needed here.
```

Comment:
Could we give an exmaple where preemption is needed?

Reply:
... or clarify why we do not need preemption here (single model partition changing this variable?)...

Reply:
... the HTML around the snippet says "In this example it is actually not needed since CallbackIntermediateUpdate is only called from one model partition." This is supposed to address your remarks. It's not in the same file so maybe that's why you missed it...